pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -175,26 +255,98 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
 	return reconcile.Result{}, nil
 }
 
-func fetchOvirtVM(con *ovirtsdk.Connection, vmSpec *v2vv1alpha1.VirtualMachineImportOvirtSourceVMSpec) (*ovirtsdk.Vm, error) {
-	// Id of the VM specified:
-	if vmSpec.ID != nil {
-		response, err := con.SystemService().VmsService().VmService(*vmSpec.ID).Get().Send()
-		if err != nil {
-			return nil, err
+func ensureDVSecretExists(client client.Client, instance *v2vv1alpha1.VirtualMachineImport, namespace string, ovirtSecretDataMap map[string]string) error {
+	secretObj := &corev1.Secret{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: ovirtSecret, Namespace: namespace}, secretObj)
+	if err != nil && errors.IsNotFound(err) {
+		dvSecret := createDVSecret(ovirtSecretDataMap, instance)
+		return client.Create(context.TODO(), dvSecret)
+	}
+	return err
+}
+
+func ensureDVConfigmapExists(client client.Client, instance *v2vv1alpha1.VirtualMachineImport, namespace string, ovirtSecretDataMap map[string]string) error {
+	cm := &corev1.ConfigMap{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: ovirtConfigmap, Namespace: namespace}, cm)
```

Comment:
Shouldn't this name  be unique per ovirt provider instance and not same name for all ovirt providers since its content is provider-specific ?

Reply:
Not sure if per ovirt provider. Should we also take into account direct execution of the API not using UI? Definitelly we have to create different cm/secret for different credentials, so here using name as constant is wrong.

Reply:
I meant we can use same one for a single ovirt provider. But the name should vary for other ovirt providers.